auditlog/events/client.go

```diff
@@ -0,0 +1,92 @@
+package events
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/workos-inc/workos-go/auditlog"
+)
+
+const (
+	userAgent  = "workos-go/0.1"
+	eventsPath = "/events"
+)
+
+// List returns a list of events.
+func List(params auditlog.ListRequestParams) *Iter {
+	return &Iter{auditlog.GetIter(params, func(params auditlog.ListRequestParams) ([]interface{}, auditlog.ListMeta, error) {
+		list := auditlog.EventList{}
+		list, err := FindAll(params)
+
+		ret := make([]interface{}, len(list.Data))
+		for i, v := range list.Data {
+			ret[i] = v
+		}
+
+		return ret, list.ListMeta, err
```

Comment:
It wasn't immediately clear why `[]interface{}` is being returned instead of `*auditlog.Event`, but based on the implementation it looks like you're building an array of `*auditlog.Event` structs.

Reply:
As I mentioned before, I tried to make `Iter` reusable with other entities that require pagination. If there is a new entity, all the developer has to do is implement a new `Query` function.

Reply:
Since `interface{}` is the return type you should be able to just pass `list.Data` directly since it matches.

Reply:
Is it not possible to return `list.Data` here? Since the elements in `list.Data` all implement the Iterable interface it should work as:

```suggestion
		return list.Data, list.ListMeta, err
```